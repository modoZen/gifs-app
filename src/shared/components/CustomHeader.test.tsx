import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import { CustomHeader } from "./CustomHeader";

describe("CustomHeader tests", () => {
  it("should match snapshoot", () => {
    const title = "Buscador de Gifs";

    const { container } = render(<CustomHeader title={title} />);

    expect(container).toMatchSnapshot();
  });

  it("should render the description when provided", () => {
    const title = "Buscador de Gifs";
    const description = "Descubre y comparte el gif perfecto";

    render(<CustomHeader title={title} description={description} />);

    expect(screen.getByText(description)).toBeInTheDocument();
  });

  it("should not render description when not provided", () => {
    const title = "Buscador de Gifs";

    render(<CustomHeader title={title} />);

    expect(screen.queryByRole("paragraph")).not.toBeInTheDocument();
  });
});
