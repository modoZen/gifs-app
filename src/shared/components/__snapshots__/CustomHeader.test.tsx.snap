// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CustomHeader tests > should match snapshoot 1`] = `
<div>
  <div
    class="content-center"
  >
    <h1>
      Buscador de Gifs
    </h1>
  </div>
</div>
`;
